boards/arm/lpcxpresso55s69/CMakeLists.txt

```diff
@@ -33,4 +36,65 @@ if (CONFIG_BUILD_WITH_TFM)
 	if(${SREC_CAT} STREQUAL SREC_CAT-NOTFOUND)
 		message(FATAL_ERROR "'srec_cat' not found. Please install it, or add it to $PATH.")
 	endif()
+
+	if (CONFIG_TFM_BL2_FALSE)
+		#merge tfm_s and zephyr_ns to a single image
+		#offset needs to be the same value as flash_layout.h in TFM
+		set_property(GLOBAL APPEND PROPERTY extra_post_build_commands
+			COMMAND ${SREC_CAT}
+			ARGS ${CMAKE_BINARY_DIR}/tfm/bin/tfm_s.bin -Binary
+				${CMAKE_BINARY_DIR}/zephyr/${KERNEL_BIN_NAME} -Binary
+				-offset 0x40000
+				-o ${CMAKE_BINARY_DIR}/tfm_merged.bin -Binary
+		)
+	else()
```

Comment:
This will be executed if `CONFIG_TFM_BL2_TRUE=y` but also if `CONFIG_TFM_BL2_CONFIG_FILE_DEFAULT=y` and the TFM config file has `TFM_BL2=False`.

I expect things will break in that case.

Should this have been an `elseif()` ?
And how is the `CONFIG_TFM_BL2_CONFIG_FILE_DEFAULT=y` supposed to be handled ?

Reply:
@tejlmand i've updated CMakeLists.txt with kconfig values and added a comment where it was not possible.

> This will be executed if CONFIG_TFM_BL2_TRUE=y but also if CONFIG_TFM_BL2_CONFIG_FILE_DEFAULT=y and the TFM config file has TFM_BL2=False.
 
This is not the case, `TFM_BL2` is set according to Zephyr's KConfig.
`CONFIG_TFM_BL2_CONFIG_FILE_DEFAULT=y` is the default and how it was tested most of the time.
`CONFIG_TFM_BL2_TRUE=y` was tested and has the same result.

`CONFIG_TFM_BL2_CONFIG_FILE_DEFAULT=y` does not set `-DBL2` in TFM which meas, we use board's default in TFM. (All boards have `BL2=ON` as default).
`CONFIG_TFM_BL2_TRUE/FALSE` forces the build with `-DBL2=ON/OFF`.

Reply:
Thanks for the explanation.

> CONFIG_TFM_BL2_CONFIG_FILE_DEFAULT=y does not set -DBL2 in TFM which meas, we use board's default in TFM. (All boards have BL2=ON as default).

Maybe worth mentioning this in a comment ?